GooglePlayInstant/Editor/PlayInstantAssetBundleUrlVerifierWindow.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace GooglePlayInstant.Editor
+{
+    /// <summary>
+    /// Window that verifies AssetBundles from given URLs.
+    /// </summary>
+    public class PlayInstantAssetBundleUrlVerifierWindow : EditorWindow
+    {
+        private static bool _assetBundleDownloadIsSuccessful;
+        private static string _assetBundleUrl;
+        private static long _responseCode;
+        private static string _errorDescription;
+        private static double _numOfMegabytes;
+
+        private const int FieldMinWidth = 170;
+
+        /// <summary>
+        /// Creates a dialog box that details the success or failure of an AssetBundle retrieval from a given assetBundleUrl.
+        /// </summary>
+        public static void ShowWindow()
+        {
+            if (string.IsNullOrEmpty(QuickDeployConfig.Config.assetBundleUrl))
+            {
+                Debug.LogError("Problem with verifying AssetBundle: AssetBundle URL cannot be empty.");
+            }
+            else
+            {
+                // Set AssetBundle url in a private variable so that information displayed in window is consistent with
+                // the url that this was called on. 
+                _assetBundleUrl = QuickDeployConfig.Config.assetBundleUrl;
+                UpdateAssetBundleVerificationInfoWindow();
+                GetWindow(typeof(PlayInstantAssetBundleUrlVerifierWindow), true, "Play Instant AssetBundle Verify");
+            }
+        }
+
+        //TODO: Support Unity 5.6.0+
+        private static void UpdateAssetBundleVerificationInfoWindow()
+        {
+            var www = UnityWebRequestAssetBundle.GetAssetBundle(_assetBundleUrl);
+            www.SendWebRequest();
+            while (!www.isDone)
```

Comment:
This looks like a busy wait on the main thread. Does this freeze the UI? I guess you can't use coroutines/yield in the editor...

Is an alternative just return if !www.isDone and wait for OnGUI to call back on the next frame?

You'd probably need to make your "var www" a field and start the web request on Awake() and when they click Refresh.

Reply:
onGUI() doesn't feel like the right place to get a callback because it only updates when there's been a change to the GUI in some way (when you move your mouse over it, when you move the dialog box, etc). It feels like a bit of an unsettling place to put this callback.

I did experiment with this though, and it caused Unity to crash instantly and consistently. Experimenting with this currently.

Reply:
Back to my question: does the UI freeze while the file download is happening?

Based on the answer there we can discuss some options.